server/core/src/reServerLib.cpp

```diff
@@ -409,7 +409,15 @@ runQueuedRuleExec( rcComm_t *rcComm, reExec_t *reExec,
     int thrInx;
 
     inx = -1;
-    while ( time( NULL ) <= endTime && ( thrInx = allocReThr( reExec ) ) >= 0 ) {
+    while ( time( NULL ) <= endTime ) {
+	if ( ( thrInx = allocReThr( reExec ) ) < 0 ) {
```

Comment:
Up to here there is no real change.  Follow this up with ``break;`` to get the original behaviour.